### PR TITLE
Bug Fix and Perf Tweak #minor

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

# TL;DR
 - Building a cache from Flyteadmin executions, should stop querying
flyteadmin, if the workflow is in terminal state
 - Default for Max streak length can be optimized to 8 (number of rounds
needed to complete one workflow start to end, with inmemory execution)
 - Other defaults set that have been hardened over time.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/882

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]

**Review by katrogan — DISMISSED** [review]

**Review by katrogan — APPROVED** [review]

File: pkg/controller/config/config.go [hunk 1]
```diff
@@ -15,22 +45,38 @@ var (
...
+		},
+		MaxWorkflowRetries:  10,
 		MaxDatasetSizeBytes: 10 * 1024 * 1024,
 		Queue: CompositeQueueConfig{
-			Type: CompositeQueueSimple,
+			Type: CompositeQueueBatch,
+			BatchingInterval: config.Duration{
+				Duration: time.Second,
+			},
+			BatchSize: -1,
```
Comment (EngHabu) [inline pkg/controller/config/config.go:62]: What does -1 mean?

Comment (kumare3) [reply pkg/controller/config/config.go:62]: all elements in the batch (infinite) this is what we have set everywhere
